### ISSUE TITLE
RS0046 should ignore implementations with matching names

### ISSUE THREAD

**Issue by sharwell** [dotnet/roslyn-analyzers#3814]
#### Analyzer package

Roslyn.Diagnostics.Analyzers

#### Package Version

3.3.0-beta1.20262.5

#### Diagnostic ID

RS0046

#### Repro steps

```csharp
abstract class Base
{
  public abstract void Method(TextSpan? selectionOpt);
}

class Derived : Base
{
  public override void Method(TextSpan? selectionOpt) { }
}
```

#### Expected behavior

RS0046 is reported on `Base.Method`.

#### Actual behavior

RS0046 is reported on both `Base.Method` and `Derived.Method`.

#### Additional details

* This exclusion should apply to both interface implementations and overriding members.
* The exclusion only applies when the implementation parameter name is the _same_ as the base parameter name. If the parameter names do not match, the implementation will report a warning if applicable.

**Comment by Evangelink** [comment]
I feel so stupid to have missed this obvious case...
